bokeh/charts/_chart.py

```diff
@@ -0,0 +1,261 @@
+"""This is the Bokeh charts interface. It gives you a high level API to build
+complex plot is a simple way.
+
+This is the main Chart class which is able to build several plots using the low
+level Bokeh API. It setups all the plot characteristics and lets you plot
+different chart types, taking OrderedDict as the main input. It also supports
+the generation of several outputs (file, server, notebook).
+"""
+#-----------------------------------------------------------------------------
+# Copyright (c) 2012 - 2014, Continuum Analytics, Inc. All rights reserved.
+#
+# Powered by the Bokeh Development Team.
+#
+# The full license is in the file LICENSE.txt, distributed with this software.
+#-----------------------------------------------------------------------------
+
+#-----------------------------------------------------------------------------
+# Imports
+#-----------------------------------------------------------------------------
+
+import numpy as np
+
+from ._chart_options import ChartOptions
+from ..browserlib import view
+from ..document import Document
+from ..embed import file_html
+from ..models import (
+    CategoricalAxis, DatetimeAxis, Grid, Legend, LinearAxis, Plot)
+from ..plotting import DEFAULT_TOOLS
+from ..plotting_helpers import _process_tools_arg
+from ..resources import INLINE
+from ..session import Session
+from ..utils import publish_display_data
+
+#-----------------------------------------------------------------------------
+# Classes and functions
+#-----------------------------------------------------------------------------
+
+def _make_method(prop_name):
+    def method(self, value):
+        setattr(self._options, prop_name, value)
+        return self
+    method.__doc__ = """ Chained method for %s option.
+    """ % prop_name
+    return method
+
+def chained_options(opts_type):
+    def wrapper(cls):
+        orig_init = cls.__init__
+
+        cls_props = set(cls.properties())
+
+        def __init__(self, *args, **kwargs):
+            self._options = opts_type(**kwargs)
+            orig_init(self)
+
+        cls.__init__ = __init__
+
+        for prop_name in opts_type.properties():
+            if prop_name not in cls_props:
+                setattr(cls, prop_name, _make_method(prop_name))
+
+        return cls
+    return wrapper
+
+@chained_options(ChartOptions)
```

Comment:
THIS IS AWESOME! @damianavila look what @bryevdv did to make you happy [with chained methods]! :stuck_out_tongue_closed_eyes: